### PR TITLE
sdk/log/logtest: Add RecordFactory

### PR DESCRIPTION
**Pull request by pellared**
Fixes https://github.com/open-telemetry/opentelemetry-go/issues/5192.

Follows @MrAlias idea to make the implementation simpler than https://github.com/open-telemetry/opentelemetry-go/pull/5234

See test example for usage.

The benefit of NOT using a fluent builder is that all code that uses `RecordFactory` would look similar.

When using a fluent builder one could write:

```go
b := logtest.RecordBuilder{}
b = b.SetInstrumentationScope(instrumentation.Scope{Name: "myapp"})
b = b.SetBody(logapi.StringValue("foo"))
r := b.Record()
```

and someone else could write:

```go
r := logtest.RecordBuilder{}.
  SetInstrumentationScope(instrumentation.Scope{Name: "myapp"}).
  SetBody(logapi.StringValue("foo")).
  Record()
```

Moreover, fluent builder also makes it easy to make a mistake in test code. E.g. for current implementation the code below would return an empty record (sic!):

```go
b := logtest.RecordBuilder{}
b.SetInstrumentationScope(instrumentation.Scope{Name: "myapp"})
b.SetBody(logapi.StringValue("foo"))
r := b.Record()
```

### CODE REVIEW

File: sdk/log/logtest/builder.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	TraceFlags        trace.TraceFlags
+
+	Resource             *resource.Resource
+	InstrumentationScope instrumentation.Scope
+
+	DroppedAttributes int
+}
+
+// Record returns a log record.
+func (b RecordFactory) Record() sdklog.Record {
```
Comment (MrAlias) [inline sdk/log/logtest/builder.go:42]: nit: `NewRecord` might convey the creation functionality a bit better.

```suggestion
func (b RecordFactory) NewRecord() sdklog.Record {
```

Not critical as this looks good enough to start with.

**Review by MrAlias — APPROVED** [review]

File: sdk/log/logtest/builder.go (hunk 1 above)
Comment (pellared) [reply sdk/log/logtest/builder.go]: https://github.com/open-telemetry/opentelemetry-go/pull/5258/commits/2bb62ff22f29e2894b3f8aea0e186c4084b3e7a6

File: CHANGELOG.md [hunk 2]
```diff
@@ -26,6 +26,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
   See our [versioning policy](VERSIONING.md) for more information about these stability guarantees. (#5240)
 - The `go.opentelemetry.io/otel/semconv/v1.25.0` package.
   The package contains semantic conventions from the `v1.25.0` version of the OpenTelemetry Semantic Conventions. (#5254)
+- Add `RecordFactory` in `go.opentelemetry.io/otel/sdk/log/logtest` to facilitate testing the exporter and processor implementations. (#5234)
```
Comment (dmathieu) [inline CHANGELOG.md:29]:
```suggestion
- Add `RecordFactory` in `go.opentelemetry.io/otel/sdk/log/logtest` to facilitate testing the exporter and processor implementations. (#5258)
```

**Review by dmathieu — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -26,6 +26,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
   See our [versioning policy](VERSIONING.md) for more information about these stability guarantees. (#5240)
 - The `go.opentelemetry.io/otel/semconv/v1.25.0` package.
   The package contains semantic conventions from the `v1.25.0` version of the OpenTelemetry Semantic Conventions. (#5254)
+- Add `RecordFactory` in `go.opentelemetry.io/otel/sdk/log/logtest` to facilitate testing the exporter and processor implementations. (#5258)
```
Comment (pellared) [inline CHANGELOG.md:29]: We can remove this if we add it to the upcoming release.

CC @XSAM

**Review by XSAM — COMMENTED** [review]
LGTM

**Pull request merged**